pkg/operator/csi/credentialsrequestcontroller/credentials_request_controller.go

```diff
@@ -193,3 +192,38 @@ func isProvisioned(cr *unstructured.Unstructured) (bool, error) {
 
 	return provisionedValBool, nil
 }
+
+func shouldSync(cloudCredentialLister operatorv1lister.CloudCredentialLister) (bool, error) {
+	clusterCloudCredential, err := cloudCredentialLister.Get(clusterCloudCredentialName)
+	if err != nil {
+		klog.Errorf("Failed to get cluster cloud credential: %v", err)
+		return false, err
+	}
+
+	isManualMode := func() bool {
+		return clusterCloudCredential.Spec.CredentialsMode == opv1.CloudCredentialsModeManual
+	}
+
+	isAwsStsEnabled := func() bool {
+		return os.Getenv("ROLEARN") != ""
+	}
+
+	isGcpWifEnabled := func() bool {
+		poolID := os.Getenv("POOL_ID")
+		providerID := os.Getenv("PROVIDER_ID")
+		serviceAccountEmail := os.Getenv("SERVICE_ACCOUNT_EMAIL")
+		projectNumber := os.Getenv("PROJECT_NUMBER")
+
+		return poolID != "" &&
+			providerID != "" &&
+			serviceAccountEmail != "" &&
+			projectNumber != ""
+	}
```

Comment:
```suggestion
	poolID := os.Getenv("POOL_ID")
	providerID := os.Getenv("PROVIDER_ID")
	serviceAccountEmail := os.Getenv("SERVICE_ACCOUNT_EMAIL")
	projectNumber := os.Getenv("PROJECT_NUMBER")

	isGcpWifEnabled := poolID != "" &&	providerID != "" && serviceAccountEmail != "" && projectNumber != ""
```

Or make it a function outside of shouldSync.